ProcessMaker/Models/CallActivity.php

```diff
@@ -40,7 +40,20 @@ function ($self, TokenInterface $token, FlowInterface $sequenceFlow) {
                 $startEvent = $startId ? $callable->getEngine()->getStorage()->getElementInstanceById($startId) : null;
                 $dataStore = $callable->getRepository()->createDataStore();
                 // The entire data model is sent to the target
-                $dataStore->setData($token->getInstance()->getDataStore()->getData());
+                $data = $token->getInstance()->getDataStore()->getData();
+
+                // Add info about parent
+                $data['_parent'] = [
+                    'process_id' => $token->getInstance()->process_id,
+                    'node_id' => $token->node_id,
+                ];
+
+                $configString = $this->getProperty('config');
```

Comment:
What happens if this isn't JSON? Is there always a config?

Reply:
That should return null, thats why I do a check on the next line

Reply:
Oh and config is the bpmn attribute 'pm:config' which is always json if it exists